### ISSUE TITLE
A large mount of data to load and fit to train  causing Anaconda kernel died and restart

### ISSUE THREAD

**Issue by Cuids** [tensorflow/tensorflow#48425]
I‘m doing a semantic segmentation task，i have thousands of pictures and labels. The kernel will died and restart when i use anaconda to train pictures that exceed 1600, but the kernel will not died when i train 1000 pictures.But the result is so bad and i want to train more than 10000 pictures.It hint the information when i use the Pycharm:

2021-04-09 11:06:08.187912: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'cupti64_110.dll'; dlerror: cupti64_110.dll not found
2021-04-09 11:06:08.190073: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'cupti.dll'; dlerror: cupti.dll not found
2021-04-09 11:06:08.191626: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1415] function cupti_interface_->Subscribe( &subscriber_, (CUpti_CallbackFunc)ApiCallback, this)failed with error CUPTI could not be loaded or symbol could not be found.
2021-04-09 11:06:08.192304: I tensorflow/core/profiler/lib/profiler_session.cc:172] Profiler session tear down.
2021-04-09 11:06:08.192482: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1496] function cupti_interface_->Finalize()failed with error CUPTI could not be loaded or symbol could not be found.
Process finished with exit code -1073741819 (0xC0000005)

I tried to solve this problem but failed.
My system information are:
Python version: 3.8
CUDA/cuDNN version: 10.1、7.6
TensorFlow version : 2.4.1
Windows10

Fallowing is my code of loading the datasets：

import cv2
from PIL import Image
import numpy as np
import os

path = 'D:\CDS\cropped\label\_label_1_6.png'
def create_one_hot_labels(path):
    NCLASSES = 6
    img = cv2.imread(path)
    labels = np.zeros((img.shape[0], img.shape[1], NCLASSES))
    for i in range(img.shape[0]):
        for j in range(img.shape[1]):
            if (img[i, j, 0] == 0.) and (img[i, j, 1] == 0.) and (img[i, j, 2] == 0.):  #black 
                labels[i, j, 0] = 1  #to create one-hot-label making the third dimension have shape of six
            elif (img[i, j, 0] == 0.) and (img[i, j, 1] == 0.) and (img[i, j, 2] == 255.): #red
                labels[i, j, 1] = 1 
            elif (img[i, j, 0] == 0.) and (img[i, j, 1] == 255.) and (img[i, j, 2] == 0.): #green
                labels[i, j, 2] = 1
            elif (img[i, j, 0] == 255.) and (img[i, j, 1] == 0.) and (img[i, j, 2] == 0.): #blue
                labels[i, j, 3] = 1
            elif (img[i, j, 0] == 255.) and (img[i, j, 1] == 255.) and (img[i, j, 2] == 0.): #cyan
                labels[i, j, 4] = 1
            else:
                labels[i, j, 5] = 1
    labels = np.reshape(labels, (-1, NCLASSES))
    return labels

for i in range(1200):
    one_hot_label = create_one_hot_labels(paths_label[i*16])
    one_hot_label = np.array(one_hot_label).astype(np.float32)
    labels.append(one_hot_label)      #train labels

X_train = []
for j in range(1200):
    img = cv2.imread(paths_image[j*16])
    img = (np.array(img) / 255).astype(np.float32)
    X_train.append(img)           #train data

for i in range(700):
    one_hot_label = create_one_hot_labels(paths_label[i*37])
    one_hot_label = np.array(one_hot_label).astype(np.float32)
    val_labels.append(one_hot_label)

X_val = []
for j in range(700):
    img = cv2.imread(paths_image[j*37])
    img = (np.array(img) / 255.).astype(np.float32)
    X_val.append(img)        #validation data

import tensorflow as tf
import numpy as np

AUTOTUNE = tf.data.experimental.AUTOTUNE
BATCH_SIZE = 20
train_ds = tf.data.Dataset.from_tensor_slices((X_train, labels))
# val_ds = tf.data.Dataset.from_tensor_slices((X_val, val_labels))
val_ds = tf.data.Dataset.from_tensor_slices((X_val, val_labels))
train_ds = train_ds.cache()
train_ds = train_ds.apply(
  tf.data.experimental.shuffle_and_repeat(buffer_size=200))
train_ds = train_ds.batch(BATCH_SIZE).prefetch(buffer_size=AUTOTUNE)
val_ds = val_ds.batch(BATCH_SIZE)

miou = tf.keras.metrics.MeanIoU(num_classes=6)
model.compile(optimizer=keras.optimizers.Adam(3e-3),
              loss='categorical_crossentropy',
              metrics=['accuracy', tf.keras.metrics.MeanIoU(num_classes=6)])
import tensorflow as tf

EPOCHS = 500
VAL_SUBSPLITS = 5
VALIDATION_STEPS = 700//BATCH_SIZE//VAL_SUBSPLITS
STEPS_PER_EPOCH = 1200//BATCH_SIZE

model_history = model.fit(train_ds, epochs=EPOCHS,
                          steps_per_epoch=STEPS_PER_EPOCH,
                          validation_steps=VALIDATION_STEPS,
                          validation_data=val_ds,

                          callbacks=[early_stopping, tensorboard_callback, csv_log])

So my problem is if the way of loading the dataset is wrong and how can i load more than 10000 pictures to train. Or if i have the problem in the version of the configuration.

Many thanks

**Comment by UsharaniPagadala** [comment]
@Cuids 
Code shared is full of indentation errors, could you please share a colab gist with issue reported or simple stand alone indented code with all dependencies such that we can replicate the issue reported.
Thanks.
